src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1121,16 +1120,19 @@ def dbapi(
             if isinstance(custom_schema, str):
                 struct_schema = type_string_to_type_object(custom_schema)
                 if not isinstance(struct_schema, StructType):
-                    logger.error(f"Invalid schema string: {custom_schema}")
                     raise ValueError(
                         f"Invalid schema string: {custom_schema}. "
                         f"You should provide a valid schema string representing a struct type."
+                        'For example: "id INTEGER, int_col INTEGER, text_col STRING".'
                     )
             elif isinstance(custom_schema, StructType):
                 struct_schema = custom_schema
             else:
-                logger.error(f"Invalid schema type: {type(custom_schema)}")
-                raise TypeError(f"Invalid schema type: {type(custom_schema)}.")
+                raise TypeError(
```

Comment:
Is ValueError better? I feel [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) is usually for python object type error.

Same question here, do we want to give users some suggestions?

Reply:
yeah, I have added some suggestion here